providers/ofrep/internal/outbound/http.go

```diff
@@ -0,0 +1,57 @@
+package outbound
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	of "github.com/open-feature/go-sdk/openfeature"
+)
+
+const ofrepV1 = "/ofrep/v1/evaluate/flags/"
+
+type AuthCallback func() (key string, value string)
+
+// Outbound client for http communication
+type Outbound struct {
+	auth    AuthCallback
+	baseURI string
+
+	client http.Client
+}
+
+func NewOutbound(baseUri string, callback AuthCallback) *Outbound {
+	client := http.Client{
```

Comment:
intermediate comment, though WIP:
Should expose the http client to the provider config ?
so consumer can have control of timeout, proxy etc'.

Reply:
Yeah, we can give some configurations here. I will add these with next commits

Reply:
@liran2000 I avoided adding any extra configurations in this first PR. I think we will need to add support for self-signed certificates and custom timeouts. But I think I will keep this as a follow-up PR

Reply:
I would also prefer having the possibility of passing through my own http client instead of providing a subset of its configuration. This will limit our maintenance effort to support all possible configurations.

Reply:
@thisthat this's a really good suggestion. I added `WithClient(*http.Client)` option with https://github.com/open-feature/go-sdk-contrib/pull/477/commits/a9e7df81a2a256015c504108e4343baa78ab62f8 allowing to override the http client with a custom one :)

Reply:
@liran2000 commit https://github.com/open-feature/go-sdk-contrib/pull/477/commits/a9e7df81a2a256015c504108e4343baa78ab62f8 fulfills your suggestion :)